src/main/java/org/opensearch/knn/index/codec/transfer/OffHeapFloatVectorTransfer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import org.apache.commons.lang.StringUtils;
+import org.opensearch.knn.index.vectorvalues.KNNFloatVectorValues;
+import org.opensearch.knn.jni.JNICommons;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Transfer float vectors to off heap memory.
+ */
+public final class OffHeapFloatVectorTransfer extends OffHeapQuantizedVectorTransfer<float[], float[]> {
+
+    public OffHeapFloatVectorTransfer(KNNFloatVectorValues vectorValues, Long batchSize) throws IOException {
+        super(vectorValues, batchSize, (vector, state) -> vector, StringUtils.EMPTY, DEFAULT_COMPRESSION_FACTOR);
+    }
+
+    public OffHeapFloatVectorTransfer(KNNFloatVectorValues vectorValues) throws IOException {
+        this(vectorValues, null);
+    }
+
+    @Override
+    protected float[] copyVector(float[] vector) {
+        return Arrays.copyOf(vector, vector.length);
+    }
+
+    @Override
+    protected long transfer(List<float[]> vectorsToTransfer, long bytesPerVector, boolean append) throws IOException {
+        return JNICommons.storeVectorData(
+            getVectorAddress(),
+            vectorsToTransfer.toArray(new float[][] {}),
+            this.batchSize * bytesPerVector,
```

Comment:
I want to know how bytesPerVector * batchSize gives the initialCapacity for the memory location created in C++ layer? can you clarify few things here.

Reply:
may be the variable name is not correct. In c++, it is representing what is the length of the 1D array we should have which is calculated as `dimensions per vector * batchSize` `bytesPerVector` tells me that if I have a float[] of 5 dimension, then this value should be 5 * 4 = 20.

Please check what it is. I think variable name is not correct.

Reply:
> In c++, it is representing what is the length of the 1D array we should have which is calculated as dimensions per vector * batchSize

Good catch, This is reserving way more capacity than required